tests/core/test_accounts.py

```diff
@@ -360,6 +343,52 @@ def test_eth_account_sign_data_with_intended_validator(message, message_type, ac
     assert new_addr == account.address
 
 
+@pytest.mark.parametrize(
+    'message, message_type',
+    (
+        ("hello world", "text"),
+        (b'hello world', "primitive"),
+        ("68656c6c6f20776f726c64", "hexstr"),
+    )
+)
+def test_eth_short_account_address_sign_data_with_intended_validator(message, message_type, acct):
+    account = acct.create()
+    kwargs = {message_type: message}
+
+    address_in_bytes = to_bytes(hexstr=account.address)
+    # Test for all lengths of addresses < 20 bytes
+    for i in range(1, 21):
+        with pytest.raises(TypeError):
+            # Raise TypeError if the address is less than 20 bytes
+            defunct_hash_message(
+                **kwargs,
+                signature_version=b'\x00',
+                version_specific_data=to_hex(address_in_bytes[:-i]),
+            )
+
+
+@pytest.mark.parametrize(
+    'message, message_type',
+    (
+        ("hello world", "text"),
+        (b'hello world', "primitive"),
+        ("68656c6c6f20776f726c64", "hexstr"),
```

Comment:
A good way to do this to reduce the repitition would be a parametrized fixture.

```python
@pytest.fixture(params=('text', 'primative', 'hexstr'))
def signature_kwargs(request):
    if request.params == 'text':
        return {"text": "hello world"}
    ...
```

Then in you tests you can just include `signature_kwargs` in the arguments and the test will run once for each of the fixture params.